Docs cni plugin (#14009)
hashicorp/consul: website/content/docs/connect/transparent-proxy.mdx

When transparent proxy is disabled, you must manually specify several configurations so that your applications can communicate with other services in the mesh:

* Explicitly configure upstream services by specifying a local port to access them.
* Change application to access `localhost:<chosen port>`.
* Configure applications to only listen on the loopback interface to prevent unauthorized traffic from bypassing the mesh.
    
The following diagram shows how traffic flows through the mesh without transparent proxy enabled:

![Diagram demonstrating that without transparent proxy, applications must "opt in" to connecting to their dependencies through the mesh](/img/consul-connect/without-transparent-proxy.png)

Transparent proxy is available for Kubernetes environments. As part of the integration with Kubernetes, Consul registers Kubernetes Services, injects sidecar proxies, and enables traffic redirection.

## Requirements

Your network must meet the following environment and software requirements to use transparent proxy.

* Transparent proxy is available for Kubernetes environments.
* Consul 1.10.0+

> * Consul Helm chart 0.32.0+. If you want to use the Consul CNI plugin to redirect traffic, Helm chart 0.47.0+ is required. Refer to [Enable the Consul CNI plugin](#enable-the-consul-cni-plugin) for additional information.

Author: ```suggestion
* Consul Helm chart 0.32.0+. If you want to use the Consul CNI plugin to redirect traffic, Helm chart 0.48.0+ is required. Refer to [Enable the Consul CNI plugin](#enable-the-consul-cni-plugin) for additional information.
```

Reviewer: This looks good

> * Consul on Kubernetes (`consul-k8s`) 0.26.0+

Reviewer: I believe this was to describe how only the CNI Plugin is supported in 0.48.0 and above. @trujillo-adam I would update the versions across this PR to 0.48.0 to indicate the version the CNI plugin was introduced. Thanks!

> * Consul on Kubernetes (`consul-k8s`) 0.26.0+

Author: The only new information I added was the bit about which version of the helm chart you need to to use the plugin -- the rest is existing information that I reformatted for readability. I don't think we changed the `consul-k8s` requirement to use TP, did we? Should that requirement actually be `consul-k8s-control-plane`, instead?

> * Consul on Kubernetes (`consul-k8s`) 0.26.0+

Reviewer: So I believe we can remove that last bullet because we have versioned docs and 0.26.0 was before 1.11.0 (we support n-2). Instead of Consul Helm chart 0.32.0+ you can just say Consul on Kubernetes 0.32.0+ since at that time we versioned everything with the same version (CLI, control plane , etc)

> * Consul on Kubernetes (`consul-k8s`) 0.26.0+

Author: ```suggestion
```

[consul.hashicorp.com/connect-service-upstreams](/docs/k8s/annotations-and-labels#consul-hashicorp-com-connect-service-upstreams) annotation. 
The following example configures the service to dial an upstream service called `my-service` in another Kubernetes cluster on port `1234`:

```yaml
"consul.hashicorp.com/connect-service-upstreams": "my-service:1234"
```

You do not need to configure services to explicitlly dial upstream services if your Consul clusters are connected with a [peering connection](/docs/connect/cluster-peering). 

## Usage

When transparent proxy is enabled, traffic sent to [KubeDNS](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/) 
or Pod IP addresses is redirected through the proxy. You must use a selector to bind Kubernetes Services to Pods as you define Kubernetes Services in the mesh. 
The Kubernetes Service name must match the Consul service name to use KubeDNS. This is the default behavior unless you have applied the `consul.hashicorp.com/connect-service` 
Kubernetes annotation to the service pods. The annotation overrides the Consul service name.

> Consul configures traffic redirection for each Pod bound to the Kubernetes Service with `iptables` rules. The rules direct all inbound and outbound traffic through an inbound and outbound listener on the sidecar proxy. Consul configures the proxy to route traffic to the appropriate upstream services based on [service

Author: ```suggestion
Consul configures `iptables` rules for each Pod bound to the Kubernetes Service to redirect all inbound and outbound traffic through an inbound and outbound listener on the sidecar proxy. Consul configures the proxy to route traffic to the appropriate upstream services based on [service
```

Reviewer: I like the original one better, they almost mean the same thing but traffic redirection is a good term for what we do with transparent proxy.

> Consul configures `iptables` rules for each Pod bound to the Kubernetes Service to redirect all inbound and outbound traffic through an inbound and outbound listener on the sidecar proxy. Consul configures the proxy to route traffic to the appropriate upstream services based on [service

Author: ```suggestion
Consul configures redirection for each Pod bound to the Kubernetes Service using  `iptables` rules. The rules redirect all inbound and outbound traffic through an inbound and outbound listener on the sidecar proxy. Consul configures the proxy to route traffic to the appropriate upstream services based on [service
```
